internal/service/datazone/environment_profile.go

```diff
@@ -0,0 +1,361 @@
+	plan.AwsAccountRegion = flex.StringToFramework(ctx, out.AwsAccountRegion)
+	plan.CreatedAt = flex.TimeToFramework(ctx, out.CreatedAt)
+	plan.CreatedBy = flex.StringToFramework(ctx, out.CreatedBy)
+	plan.Description = flex.StringToFramework(ctx, out.Description)
+	plan.DomainIdentifier = flex.StringToFramework(ctx, out.DomainId)
+	plan.EnvironmentBlueprintIdentifier = flex.StringToFramework(ctx, out.EnvironmentBlueprintId)
+	plan.Id = flex.StringToFramework(ctx, out.Id)
+	plan.Name = flex.StringToFramework(ctx, out.Name)
+	plan.ProjectIdentifier = flex.StringToFramework(ctx, out.ProjectId)
+	plan.UpdatedAt = flex.TimeToFramework(ctx, out.UpdatedAt)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourceEnvironmentProfile) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.Meta().DataZoneClient(ctx)
+	o := flex.AutoFlexOptions{}
+	o.AddIgnoredField("user_parameters")
+
+	// TIP: -- 2. Fetch the state
+	var state dEnvProfile
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findEnvironmentProfileByID(ctx, conn, state.Id.ValueString(), state.DomainIdentifier.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	state.AwsAccountId = flex.StringToFramework(ctx, out.AwsAccountId)
+	state.AwsAccountRegion = flex.StringToFramework(ctx, out.AwsAccountRegion)
+	state.CreatedAt = flex.TimeToFramework(ctx, out.CreatedAt)
+	state.CreatedBy = flex.StringToFramework(ctx, out.CreatedBy)
+	state.Description = flex.StringToFramework(ctx, out.Description)
+	state.DomainIdentifier = flex.StringToFramework(ctx, out.DomainId)
+	state.EnvironmentBlueprintIdentifier = flex.StringToFramework(ctx, out.EnvironmentBlueprintId)
+	state.Id = flex.StringToFramework(ctx, out.Id)
+	state.Name = flex.StringToFramework(ctx, out.Name)
+	state.ProjectIdentifier = flex.StringToFramework(ctx, out.ProjectId)
+	state.UpdatedAt = flex.TimeToFramework(ctx, out.UpdatedAt)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceEnvironmentProfile) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	conn := r.Meta().DataZoneClient(ctx)
+
+	var plan, state dEnvProfile
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if plan.EnvironmentBlueprintIdentifier.Equal(state.EnvironmentBlueprintIdentifier) &&
+		plan.ProjectIdentifier.Equal(state.ProjectIdentifier) && plan.DomainIdentifier.Equal(state.DomainIdentifier) {
```

Comment:
Should this be inverted?

```suggestion
	if !plan.EnvironmentBlueprintIdentifier.Equal(state.EnvironmentBlueprintIdentifier) ||
		!plan.ProjectIdentifier.Equal(state.ProjectIdentifier) || !plan.DomainIdentifier.Equal(state.DomainIdentifier) {
```